pkg/service/service_test.go

```diff
@@ -861,3 +861,61 @@ func (m *ManagerTestSuite) TestRestoreServiceWithTerminatingBackends(c *C) {
 	}
 	c.Assert(m.lbmap.DummyMaglevTable[uint16(id1)], Equals, len(backends1))
 }
+
+// l7 load balancer service should be able to override a ClusterIP service with same
+// frontend, but reverse should produce an error. Also, it should not override
+// any other type besides itself or clusterIP type.
+func (m *ManagerTestSuite) TestL7LoadBalancerServiceOverride(c *C) {
```

Comment:
The test isn't doing what the text above describes. It's adding services of the same type `SVCTypeClusterIP` back-to-back instead of `SVCTypeL7LoadBalancer`? Anyway I don't see any code that tackles this - 

>l7 load balancer service should be able to override a ClusterIP service with same
// frontend, but reverse should produce an error.

Reply:
I had neglected to update the comment as the design evolved. It now reads:
```
// l7 load balancer service should be able to override any service type
// (Cluster IP, NodePort, etc.) with same frontend.
```